.github/workflows/update_python_dependencies.yml

```diff
@@ -0,0 +1,77 @@
+# Licensed to the Apache Software Foundation (ASF) under one
+# or more contributor license agreements.  See the NOTICE file
+# distributed with this work for additional information
+# regarding copyright ownership.  The ASF licenses this file
+# to you under the Apache License, Version 2.0 (the
+# "License"); you may not use this file except in compliance
+# with the License.  You may obtain a copy of the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing,
+# software distributed under the License is distributed on an
+# "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+# KIND, either express or implied.  See the License for the
+# specific language governing permissions and limitations
+# under the License.
+
+# This workflow will update python dependencies as part of the release process
+# And commit the changes into the branch release, creating a PR into the branch 
+
+name: Update Python Depedencies
+
+on:
+  workflow_dispatch:
+    inputs:
+      BRANCH_NAME:
+        description: Name of the new branch that will contain the changes 
+        required: true
```

Comment:
Should we be doing this on a cron schedule (maybe weekly) instead of manually? I'd still leave the option of doing it manually, but we would probably benefit from doing it more often.

@tvalentyn @kennknowles any concerns with doing it more frequently?

Reply:
once a week sounds good. will it send a PR for review?

Reply:
Yes (if there are changes, it no ops if there are no changes)

Reply:
sg

Reply:
Done, added a cron trigger to execute the workflow once a week